### ISSUE TITLE
Two MS5803 sensors on I2C bus

### ISSUE THREAD

**Issue by trafidi96** [millerlp/MS5803_02#2]
Thank you for providing this code. 

I wanted to use multiple MS5803-02B sensors on one I2C bus, but no matter the method there seems to be always a systematic offset in the pressure and temperature reading of one of the sensors. Here are the ways tried, they both produced a similar output:

1- Using a multiplexer (TCA9549A) with the original library and set up provided: offset in pressure and temperature in one of the sensors observed, but the reading changes when the respective sensor is pressurized etc.

2- Tried modifying the code so that the address can be modified from the constructor, and included an address variable (private) in the header file. (1) Tested the sensors individually by passing 0x76 and 0x77 in a modified version of the example file: each worked as expected. (2) Tested the sensor working simultaneously: got the same issue as seen with multiplexer. 

Could it be a bit shift due to some addressing issue? How do you think I can resolve it? I would really appreciate the help. Thank you

**Comment by millerlp** [comment]
That's strange behavior. To be clear, if you test each sensor individually on its own, do they show the same pressure readings as each other on the bench, so there is no evidence of a factory offset/miscalibration in one of the two sensors? 

Second, is it possible that your test code is only grabbing the sensor coefficients from one sensor and then re-using those coefficients on the other sensor, producing an erroneous output for that other sensor? Line 107 in MS5803_02.cpp is where the sensor coefficients are being stored in an array, and then they get used in lines 163, 167, 204, 205 in the readSensor() function to calculate temperature and pressure. Those sensor coefficients are likely to be different among individual sensors, and so they should be kept with the appropriate sensor.

**Comment by trafidi96** [comment]
Yes, I verified this with several sensors, and they all produce comparable data. I thought that the constant might be the issue, and still think it is related. The only evidence I have against that is that when the sensors are initialized using sensor1.initializeMS_5803() and automatically print out the constants, each prints its own coefficients. This would mean that each sensor is associated with its coefficients. However, there is a chance that there is a conflict between the constants. Now I am not sufficiently trained in CS but I was thinking that the "Static" prefix would mean that the variable belongs to the class and  not the instance, but that still would not explain what is happening with the multiplexer.

**Comment by millerlp** [comment]
I'll bet it has something to do with the sensor coefficients variable being public. In this [alternate MS5803 library](https://github.com/EnviroDIY/MS5803), they keep the coefficients as a private variable. You may want to give that one a try. It also lets you directly specify the I2C address in the class definition.

**Comment by trafidi96** [comment]
Thank you for this suggestion the library you suggested seem to work fine with the I2C addressing. I haven't tried designating the variable as private in the library you developed. If end up reinvestigating that I will update the thread.
